app/vmalert/alerting.go

```diff
@@ -231,6 +233,11 @@ func (ar *AlertingRule) ExecRange(ctx context.Context, start, end time.Time) ([]
 			result = append(result, ar.alertToTimeSeries(a, s.Timestamps[i])...)
 		}
 	}
+	numActive := len(ar.alerts)
+	if limit > 0 && numActive > limit {
+		ar.alerts = map[uint64]*notifier.Alert{}
+		return nil, errors.Errorf("exec range exceeded limit of %d with %d alerts", limit, numActive)
+	}
```

Comment:
check generated alerts at one point of time  might lose part timestamp, which would be misleading

Reply:
how about add additional range limit parameter?

Reply:
> check generated alerts at one point of time might lose part timestamp, which would be misleading

true, but it would be similar to usual behaviour of vmalert when it runs as a deamon.

> how about add additional range limit parameter?

Sounds too complicated. The idea of replay was for it to be compatible with existing rules definitions. It would be inconvenient to modify the list of rules specifically for replay.
IMHO, it is better to ignore this limit for replay than adding a new config field.

Reply:
> true, but it would be similar to usual behaviour of vmalert when it runs as a deamon.
ok, i'll modify it in this way

```
The correct implementation would be to have a map with timestamp as key and number of active alerts as value. After all series were processed, the map can be checked key by key and if somewhere value exceeds the limit - log an err and remove this timestamp from results.
```